test/fetchBaseQuery.test.tsx

```diff
@@ -462,3 +457,23 @@ describe('fetchFn', () => {
     expect(request.url).toEqual(`${baseUrl}/echo?apple=fruit`);
   });
 });
+
+describe('FormData', () => {
+  test('sets the right headers when sending FormData', async () => {
+    let request: any;
+    const body = new FormData();
+    body.append('username', 'test');
+    body.append('file', new Blob([JSON.stringify({ hello: 'there' }, null, 2)], { type: 'application/json' }));
+
+    ({ data: request } = await baseQuery(
+      { url: '/echo', method: 'POST', body },
+      {
+        signal: undefined,
+        dispatch: storeRef.store.dispatch,
+        getState: storeRef.store.getState,
+      },
+      {}
+    ));
+    expect(request.headers['content-type']).not.toContain('application/json');
+  });
```

Comment:
The browser will specify the content-type of multipart/form-data with the correct boundary. Couldn't find a way to test this specifically outside of a browser, but this should be sufficient.